src/tests/nativeaot/SmokeTests/HelloWasm/HelloWasm.cs

```diff
@@ -1556,6 +1569,102 @@ private static void _CallMe(int x)
+    }
+
+    [WasmImportLinkage]
+    [DllImport("ModuleName", EntryPoint = "FunctionInModule")]
+    private static extern int CallFunctionInModule(int x);
+
+    [DllImport("ModuleName", EntryPoint = "DupImportTest"), WasmImportLinkage]
+    private static extern int WasmImportFuncDup1(int arg);
+
+    [DllImport("ModuleName", EntryPoint = "DupImportTest"), WasmImportLinkage]
+    private static extern int WasmImportFuncDup2();
+
+    [System.Runtime.InteropServices.UnmanagedCallersOnly(EntryPoint = "JustForRooting")]
+    private static void RootFuncDup(int x)
+    {
+        WasmImportFuncDup1(0);
+        WasmImportFuncDup2();
+    }
+
+    [DllImport("ModuleName1", EntryPoint = "CommonFunctionName"), WasmImportLinkage]
+    private static extern int CallFunctionInModule1(int arg);
+
+    [DllImport("ModuleName2", EntryPoint = "CommonFunctionName"), WasmImportLinkage]
+    private static extern int CallFunctionInModule2(int arg);
+
+    private static void TestSameFunctionNameInDifferentModules()
+    {
+        StartTest("Wasm import same function name from different modules test");
+        EndTest(CallFunctionInModule1(456) == 456 && CallFunctionInModule2(789) == 790);
+    }
+
+    [DllImport("StaticModule1", EntryPoint = "CommonStaticFunctionName")]
+    private static extern int CommonFunctionNameInModule1(int arg);
+
+    [DllImport("StaticModule2", EntryPoint = "CommonStaticFunctionName")]
+    private static extern int CommonFunctionNameInModule2(int arg);
+
+    private static void TestStaticPInvokeOverloadedInDifferentModules()
+    {
+        StartTest("Static PInvoke of overloaded function in different modules test");
+        EndTest(CommonFunctionNameInModule1(12) == 12 && CommonFunctionNameInModule2(34) == 34);
+    }
+
+    [DllImport("StaticModule1", EntryPoint = "CommonStaticFunctionName")]
+    private static extern int CallAbiCompatFunctionWithInt(int arg);
+
+    [DllImport("StaticModule2", EntryPoint = "CommonStaticFunctionName")]
+    private static extern uint CallAbiCompatFunctionWitUint(uint arg);
+
+    private static void TestStaticAbiCompatibleSignatures()
+    {
+        StartTest("Static imports with ABI compatible signatures");
+        EndTest(CallAbiCompatFunctionWithInt(456) == 456 && CallAbiCompatFunctionWitUint(789) == 789);
+    }
+
+    [DllImport("StaticModule1", EntryPoint = "StaticIncompatFunctionName")]
+    private static extern int CallAbiIncompatFunctionWithInt(int arg);
+
+
+    [DllImport("StaticModule2", EntryPoint = "StaticIncompatFunctionName")]
+    private static extern void CallAbiIncompatFunctionWithFloat(float arg);
+
+
+    // Compilation should produce a warning for CallAbiIncompatFunctionWithInt and CallAbiIncompatFunctionWithFloat.
+    [System.Runtime.InteropServices.UnmanagedCallersOnly(EntryPoint = "JustForRooting2")]
+    private static void RootFuncDup()
+    {
+        CallAbiIncompatFunctionWithInt(0);
+        CallAbiIncompatFunctionWithFloat(1);
+    }
+#endif
```

Comment:
Could you also add the scenario `3` test?

Reply:
Added the missing tests.